test/PositionSpec.js

```diff
@@ -59,243 +243,102 @@ describe('Position', function () {
 
       instance.setState({fakeProp: 1});
+      await wait();
 
       // Position receives new props and position should be recalculated
       expect(Position.prototype.componentWillReceiveProps)
         .to.have.been.calledOnce;
-      expect(Position.prototype.updatePosition)
+      expect(Popper.prototype.update)
         .to.have.been.calledTwice;
     });
   });
 
-  describe('position calculation', function () {
-    let mountPoint;
-
-    beforeEach(function () {
-      mountPoint = document.createElement('div');
-      document.body.appendChild(mountPoint);
-    });
-
-    afterEach(function () {
-      ReactDOM.unmountComponentAtNode(mountPoint);
-      document.body.removeChild(mountPoint);
-    });
-
-    function checkPosition(placement, targetPosition, expected) {
-      class FakeOverlay extends React.Component {
-        render() {
-          return (
-            <div style={{
-              position: 'absolute',
-              width: 200,
-              height: 200
-            }} />
-          );
-        }
-      }
-
-      class FakeContainer extends React.Component {
-        render() {
-          return (
-            <div style={{
-              position: 'relative',
-              width: 600,
-              height: 600
-            }}>
-              <div
-                ref={(c) => { this.target = c; }}
-                style={{
-                  position: 'absolute',
-                  width: 100,
-                  height: 100,
-                  ...targetPosition
-                }}
-              />
-
-              <Position
-                target={() => this.target}
-                container={this}
-                containerPadding={50}
-                placement={placement}
-              >
-                <FakeOverlay ref={(c) => { this.overlay = c; }} />
-              </Position>
-              <Position
-                target={() => this.target}
-                container={() => this}
-                containerPadding={50}
-                placement={placement}
-              >
-                <FakeOverlay ref={(c) => { this.fakeOverlay = c; }} />
-              </Position>
-            </div>
-          );
-        }
-      }
-
-      const expectedPosition = {
-        positionLeft: expected[0],
-        positionTop: expected[1],
-        arrowOffsetLeft: expected[2],
-        arrowOffsetTop: expected[3]
-      };
-
-      it('Should calculate the correct position', function() {
-        const instance = render(<FakeContainer />, mountPoint);
-
-        ['overlay', 'fakeOverlay'].forEach(function(overlayRefName) {
-          const calculatedPosition = pick(
-            instance[overlayRefName].props, Object.keys(expectedPosition)
-          );
-          expect(calculatedPosition).to.eql(expectedPosition);
-
-        });
-      });
-    }
-
-    [
-      {
-        placement: 'left',
-        noOffset: [50, 200, undefined, '50%'],
-        offsetBefore: [-200, 50, undefined, '0%'],
```

Comment:
corresponding old...

though note that the target position is a bit different now

the old impl would only apply the padding to cross-axis positioning. this on the other hand applies the padding to same-axis, mostly